client.go

```
@@ -27,6 +27,44 @@ import (
 	"github.com/gopcua/opcua/uasc"
 )
 
+// FindServers returns the Servers known to a Server or Discovery Server.
+func FindServers(ctx context.Context, endpoint string, opts ...Option) ([]*ua.ApplicationDescription, error) {
+	opts = append(opts, AutoReconnect(false))
+	c := NewClient(endpoint, opts...)
+	if err := c.Dial(ctx); err != nil {
+		return nil, err
+	}
+	defer func(c *Client, ctx context.Context) {
+		if err := c.Close(ctx); err != nil {
+			// TODO(sruehl): log error
+		}
+	}(c, ctx)
+	res, err := c.FindServers(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return res.Servers, nil
+}
+
+// FindServersOnNetwork returns the Servers known to a Server or Discovery Server. Unlike FindServers, this Service is only implemented by Discovery Servers.
+func FindServersOnNetwork(ctx context.Context, endpoint string, opts ...Option) ([]*ua.ServerOnNetwork, error) {
+	opts = append(opts, AutoReconnect(false))
+	c := NewClient(endpoint, opts...)
+	if err := c.Dial(ctx); err != nil {
+		return nil, err
+	}
+	defer func(c *Client, ctx context.Context) {
```

Same comment as on #674 . Just call `c.Close()` and ignore the error since it cannot be recovered anyway. The caller is interested in the root cause of the close.